api/v1beta1/machinedeployment_types.go

```diff
@@ -54,8 +54,17 @@ const (
 	// proportions in case the deployment has surge replicas.
 	MaxReplicasAnnotation = "machinedeployment.clusters.x-k8s.io/max-replicas"
 
-	// MachineDeploymentUniqueLabel is the label applied to Machines
-	// in a MachineDeployment containing the hash of the template.
+	// MachineDeploymentUniqueLabel is used to uniquely identify the Machines of a MachineSet.
```

Comment:
it's not immediately clear to me from reading this comment how the controller handles existing Machines that already have the hash of the machine template as label value

I'm sure I'll find out later in this review by reading the code, but might be worth clarifying for others who later run into this comment

Reply:
And if so, why not add a new label and keep this one for backwards compatibility?

Reply:
> it's not immediately clear to me from reading this comment how the controller handles existing Machines that already have the hash of the machine template as label value

The label will be kept the same. The only reason that we have this label is to uniquely identify a MachineSet and its Machines. The value must be unique but otherwise we never read or modify it for an existing MachineSet.

Reply:
> And if so, why not add a new label and keep this one for backwards compatibility?

I added this comment to the wrong comment thread, was meant for below

Reply:
Resolving this thread in favor the discussion at https://github.com/kubernetes-sigs/cluster-api/pull/8110#discussion_r1106560302. The other thread is also a discussion on the label and has more information.